src/index.ts

```diff
@@ -21,58 +21,84 @@ function log(msg: string) {
     console.log(`[${new Date().toISOString()}] ${msg}`)
 }
 
-async function fetchPrice(name: string): Promise<number> {
-    log(`fetching price of ${name}`)
+async function fetchCoinMarketCapPrice(name: string): Promise<number> {
+    if (!(name in marketNameToPage)) {
+        throw new Error(`${name} not found in CoinMarketCap`)
+    }
+    log(`[CoinMarketCap] fetching price of ${name}`)
     const url = `${COINMARKETCAP_ENDPOINT}/${marketNameToPage[name]}`
     const resp = await fetch(url, { method: 'GET', timeout: 10 * 1000 })
     const root = parse(await resp.text())
     const priceStr = root.querySelector('div.priceValue ').text
     const price = parseFloat(priceStr.substr(1).replace(',', ''))
-    log(`price: ${price}`)
+    log(`[CoinMarketCap] price: ${price}`)
     return price
 }
 
 async function fetchFtxPrice(name: string): Promise<number> {
+    name = name.toUpperCase()
     log(`[FTX] fetching price of ${name}`)
     const url = `${FTX_ENDPOINT}/${name}/USD`
     const resp = await fetch(url, { method: 'GET', timeout: 10 * 1000 })
-    const price = (await resp.json()).result.last
+    const result = (await resp.json()).result
+    if (!result) {
+        throw new Error(`${name} not found in FTX`)
+    }
+    const price = result.last || result.ask || result.bid
```

Comment:
for market such as `JPY/USD`, all three are `null`. We should still check if (!price) => throw Error